iotilecore/iotile/core/hw/debug/firmware_image_analyzer.py

```diff
@@ -0,0 +1,155 @@
+from iotile.core.utilities.intelhex import IntelHex
+from iotile.core.exceptions import *
+from tempfile import NamedTemporaryFile
+import os
+import subprocess
+import struct
+
+class FirmwareImageAnalyzer:
+    """A firmware image analyzer using IntelHex"""
+
+    KNOWN_FIRMWARE_NAMES = ['boot52', 'NRF52 ', 'firmPP', 'UART  ', 'firmFF']
+    SIZE_OF_CDBAPPINFO = 32
+    MAGIC_NUMBER_BYTE_OFFSET = 24
+
+    def __init__(self, firmware, magic_number=0xBAADDAAD):
+        if not firmware.endswith(".elf") and not firmware.endswith(".hex"):
+            raise ArgumentError("You must pass an ARM firmware image in elf/hex format", path=firmware)
+
+        tmpf = NamedTemporaryFile(delete=False)
+        tmpf.close()
+
+        tmp = tmpf.name
+
+        try:
+            if firmware.endswith(".elf"):
+                err = subprocess.call(['arm-none-eabi-objcopy', '-O', 'ihex', firmware, tmp])
+                if err != 0:
+                    raise ExternalError("Cannot convert elf to binary file", error_code=err)
+
+                hex_image = IntelHex(tmp)
+            else:
+                hex_image = IntelHex(firmware)
+
+            self._hex_image = hex_image
+        finally:
+            os.remove(tmp)
+
+        self.cdb_app_info = None
+        self.min_addr = self._hex_image.minaddr()
+        self.max_addr = self._hex_image.maxaddr()
+        self.num_addresses = len(self._hex_image.addresses())
+        self.segments = self._hex_image.segments()
+        self.memory_size = self._hex_image.get_memory_size()
+        self.magic_number = magic_number
+
+    def set_firmware_magic_number(self, magic_number):
+        """Sets the magic number to search a CDB app info block by
+        """
```

Comment:
Single line comments should have closing quotes on the same line

Reply:
Fixed